ext/java/org/msgpack/jruby/ExtensionRegistry.java

```diff
@@ -82,25 +82,23 @@ public IRubyObject lookupUnpackerByTypeId(int typeId) {
      * Objects of type Integer (Fixnum, Bignum), Float, Symbol and frozen
      * String have no singleton class and raise a TypeError when trying to get
      * it.
-     *
-     * Since all but symbols are already filtered out when reaching this code
-     * only symbols are checked here.
      */
-    if (!(object instanceof RubySymbol)) {
-      lookupClass = object.getSingletonClass();
-      pair = fetchEntryByModule(lookupClass);
-      if (pair != null) {
-          return pair;
-      }
+    lookupClass = object.getMetaClass();
```

Comment:
JRuby has similar APIs:

```java
    /**
     * if exist return the meta-class else return the type of the object.
     *
     */
    @Override
    public final RubyClass getMetaClass() {
        return metaClass;
    }

    // ....
 
       /**
     * @see org.jruby.runtime.builtin.IRubyObject#getType()
     */
    @Override
    public final RubyClass getType() {
        return metaClass.getRealClass();
    }
```

So we can use the same strategy, first `getMetaClass` and later `getType`, then check wether we got the same object or not.